### ISSUE TITLE
`resin devices supported` command does not return the list of supported device types

### ISSUE THREAD

**Issue by josephroberts** [resin-io/resin-cli#394]
The docs state the `resin devices supported` command should return the list of device types but it doesn't. I am unsure how to find out the proper type for a RPi3

```
➜  database resin devices supported
Usage: resin [COMMAND] [OPTIONS]

Run the following command to get a device started with Resin.io

  $ resin quickstart

If you need help, or just want to say hi, don't hesitate in reaching out at:

  GitHub: https://github.com/resin-io/resin-cli/issues/new
  Gitter: https://gitter.im/resin-io/chat

Primary commands:

    help [command...]                   show help                                                    
    quickstart [name]                   getting started with resin.io                                
    login                               login to resin.io                                            
    app create <name>                   create an application                                        
    apps                                list all applications                                        
    app <name>                          list a single application                                    
    devices                             list all devices                                             
    device <uuid>                       list a single device                                         
    logs <uuid>                         show device logs                                             
    sync [uuid]                         (beta) sync your changes to a device                         
    ssh [uuid]                          (beta) get a shell into the running app container of a device

Run `resin help --verbose` to list additional commands

Global Options:

    --help, -h                                                             
➜  database 
```

**Comment by jviotti** [comment]
Hi @josephroberts ,

The `device supported` command was removed a year ago (https://github.com/resin-io/resin-cli/pull/160), and its not being listed in the command list. 

Maybe worth pinging the docs maintainer to update the CLI reference?

**Comment by josephroberts** [comment]
Hey @jviotti ,

Yes we could ping @lekkas and @shaunmulligan . The docs are out of date both at https://github.com/resin-io/resin-cli/blob/master/doc/cli.markdown and at https://docs.resin.io/tools/cli/#app-create-60-name-62-

Are they linked somehow?

Why did you state "The command is not necessary and unused."?
It seems like a nice way to figure out what the correct device type should be and directly from the CLI without having to look elsewhere. I did search for a few mins and couldn't find an obvious place where the device types are listed so its probably the same for other users.

**Comment by jviotti** [comment]
Ah, its mentioned in the `app create` help command. I must have missed
that when removing it. I'll update now.

> Why did you state "The command is not necessary and unused."?  It
> seems like a nice way to figure out what the correct device type
> should be and directly from the CLI without having to look elsewhere.
> I did search for a few mins and couldn't find an obvious place where
> the device types are listed so its probably the same for other users.

Yeah, maybe we should bring it back. The CLI used to have tons of
"mini-commands" back then, and I remember we had a talk about removing
the ones that no-one needed (this command was brought up there).

Happy to add back if we all agree its useful now.

On Thu, Sep 22, 2016 at 10:21:31AM -0700, Joseph Roberts wrote:
> Hey @jviotti ,
> 
> Yes we could ping @lekkas and @shaunmulligan . The docs are out of date both at https://github.com/resin-io/resin-cli/blob/master/doc/cli.markdown and at https://docs.resin.io/tools/cli/#app-create-60-name-62-
> 
> Are they linked somehow?
> 
> Why did you state "The command is not necessary and unused."?
> It seems like a nice way to figure out what the correct device type should be and directly from the CLI without having to look elsewhere. I did search for a few mins and couldn't find an obvious place where the device types are listed so its probably the same for other users.
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/resin-io/resin-cli/issues/394#issuecomment-248969388

-- 
Juan Cruz Viotti
Software Engineer

**Comment by imrehg** [comment]
I'd recommend adding it back, if it was removed. The first issue for me with the cli is always figuring out how our devices types are named (since they are non-obvious, and often non-consistent, see pi1/2/3). I remember talking about it a few months ago. I guess I just settled in with the few types that I use and remember that, but indeed, it's troublesome to try figuring out what the name of a type is that I haven't used before. Should be all accessible in the same place, i.e. within the CLI somehow. Either as a command, or listing it in the help (hard-coded, and updated as device types are updated)

**Comment by josephroberts** [comment]
Yeah it needs adding back in IMO. See below:

```
➜  Git resin help app create
Usage: app create <name>

Use this command to create a new resin.io application.

You can specify the application type with the `--type` option.
Otherwise, an interactive dropdown will be shown for you to select from.

You can see a list of supported device types with

	$ resin devices supported

Examples:

	$ resin app create MyApp
	$ resin app create MyApp --type raspberry-pi

Options:

    --type, -t <type>                   application type                   
➜  Git resin devices supported
Usage: resin [COMMAND] [OPTIONS]

Run the following command to get a device started with Resin.io

  $ resin quickstart

If you need help, or just want to say hi, don't hesitate in reaching out at:

  GitHub: https://github.com/resin-io/resin-cli/issues/new
  Gitter: https://gitter.im/resin-io/chat

Primary commands:

    help [command...]                   show help                                                    
    quickstart [name]                   getting started with resin.io                                
    login                               login to resin.io                                            
    app create <name>                   create an application                                        
    apps                                list all applications                                        
    app <name>                          list a single application                                    
    devices                             list all devices                                             
    device <uuid>                       list a single device                                         
    logs <uuid>                         show device logs                                             
    sync [uuid]                         (beta) sync your changes to a device                         
    ssh [uuid]                          (beta) get a shell into the running app container of a device

Run `resin help --verbose` to list additional commands

Global Options:

    --help, -h                                                             
➜  Git 
```

**Comment by jviotti** [comment]
I brought back the `devices supported` command, and published v5.1.0 to NPM.
